src/main/asciidoc/_chapters/protobuf.adoc

```diff
@@ -148,3 +148,67 @@ consider extending it also in
 Going forward, we will provide a new module of common types for use
 by CPEPs that will have the same guarantees against change as does our
 public API. TODO.
+
+=== protobuf changes for hbase-3.0.0 (HBASE-23797)
+Since hadoop also shades protobuf and bumps the version to 3.x, there is no
+reason for us to stay on protobuf 2.5.0 any more.
+
+In 3.0.0, the hbase-protocol module has been purged, the CPEP implementation
+should use the protos in hbase-protocol-shaded module, and also make use of
+the shaded protobuf in hbase-thirdparty.
```

Comment:
Anything on the audience for this module? Previous it was meant to be private to hbase. Now its exposed for CPEPs.  Should we recognize that by adding annotation or noting it somewhere like here?

Reply:
We will keep protobuf version compatible for a whole major release, so maybe we could upgrade from 3.11.4 to 3.11.5 or even 3.12.0 for a new minor release line but we will never upgrade to 4.0.0. We could add this here.

Reply:
Yeah. Don't think it noted anywhere else.